instrumentation/CONTRIBUTING.md

```diff
@@ -0,0 +1,483 @@
+
+        present do
+          defined?(::Werewolf) && defined?(::ActiveSupport)
+        end
+
+        compatible do
+          Gem::Version.new(::Wereworlf::VERSION) >= MINIMUM_VERSION
+        end
+      end
+    end
+  end
+end
+
+```
+
+* The `install` block lazily requires the instrumentation handlers, which subscribe to events published by the `Werewolf` event hooks.
+* The `present` block checks if the `Werewolf` and `ActiveSupport` libraries are loaded, which it will use to subscribe to events and generate spans. It will skip the installation if those dependencies were not loaded before the instrumentation was being initialized.
+* The `compatible` block checks if the `Werewolf` library version is at least `0.1.0` and will skip it if it is not.
+* The `options` section allows you to define custom options that can be passed to the instrumentation. In this example, the `transformations` option is defined with a default value of `:omit` and a validation rule that only allows `:omit` or `:include` values.
+
+### Use the OpenTelemetry API
+
+Instrumentations are intended to be portable and usable with vendor distributions of the SDK. For this reason, you must use the OpenTelemetry API to create spans and add attributes, events, and links to spans and avoid using the OpenTelemetry SDK directly.
+
+Each instrumentation _must_ use a named tracer. Instrumentations that inherit from `OpenTelemetry::Instrumentation::Base` will get a single helper method that will automatically provide your instrumentation with a named tracer under `OpenTelemetry::Instrumentation::${Gem Name}::Instrumentation.instance.tracer`.
+
+For example, the `Werewolf` generated in the example above instrumentation is available `OpenTelemetry::Instrumentation::Werewolf::Instrumentation.instance.tracer`. You should reference this tracer in your code when creating spans like this:
+
+```ruby
+
+  OpenTelemetry::Instrumentation::Werewolf::Instrumentation.instance.tracer.start_span('transform') do
+    # code to be traced
+  end
+
+```
+
+> :warning: This tracer is not _upgradable_ before the SDK is initialized, therefore it is important that your instrumentation _always_ use stack local references of the tracer.
+
+### Use First Party Extension Points
+
+Whenever possible, use first party extension points (hooks) to instrument libraries. This ensures that the instrumentation is compatible with the latest versions of the library and that the instrumentation is maintained by the library authors. `ActiveSupport::Notifications` and `Middleware` are good examples of a first party extension points that are used by our instrumentation libraries.
+
+Monkey patching is discouraged in OpenTelemetry Ruby because it is the most common source of bugs and incompatability with the libraries we are instrumenting. If you must monkey patch, please ensure that the monkey patch is as isolated as possible and that it is clearly documented.
```

Comment:
```suggestion
Monkey patching is discouraged in OpenTelemetry Ruby because it is the most common source of bugs and incompatability with the libraries we instrument. If you must monkey patch, please ensure that the monkey patch is as isolated as possible and that it is clearly documented.
```